src/Maestro/SubscriptionActorService/PullRequestActor.cs

```diff
@@ -1038,5 +1066,219 @@ private async Task<RepositoryBranch> GetRepositoryBranch(string repo, string bra
+            List<UpdateAssetsParameters> updates,
+            InProgressPullRequest? pr)
+        {
+            // TODO https://github.com/dotnet/arcade-services/issues/3378: Support batched PRs for code flow updates
+            if (updates.Count > 1)
+            {
+                updates = [..updates.DistinctBy(u => u.BuildId)];
+                if (updates.Count > 1)
+                {
+                    _logger.LogWarning("Code flow updates cannot be batched with other updates. Will process the last update only.");
+                }
+            }
+
+            var update = updates.Last();
+
+            CodeFlowStatus? codeFlowStatus = await _codeFlowState.TryGetStateAsync();
+
+            // The E2E order of things for is:
+            // 1. We send a request to PCS and wait for a branch to be created. We note down this in the codeflow status. We set a reminder.
+            // 2. When reminder kicks in, we check if the branch is created. If not, we repeat the reminder.
+            // 3. When branch is created, we create the PR and set the usual reminder of watching a PR (common with the regular subscriptions).
+            // 4. For new updates, we only delegate those to PCS which will push in the branch.
+            if (pr == null)
+            {
+                (string targetRepository, string targetBranch) = await GetTargetAsync();
+
+                // Step 1. Let PCS create a branch for us
+                if (codeFlowStatus == null)
+                {
+                    return await RequestCodeFlowBranchAsync(update, targetBranch);
+                }
+
+                // Step 2. Wait for the branch to be created
+                IRemote remote = await _remoteFactory.GetRemoteAsync(targetRepository, _logger);
+                if (!await remote.DoesBranchExistAsync(targetRepository, codeFlowStatus.PrBranch))
+                {
+                    _logger.LogInformation("Branch {branch} for subscription {subscriptionId} not created yet. Will check again later",
+                        codeFlowStatus.PrBranch,
+                        update.SubscriptionId);
+
+                    await _pullRequestUpdateState.StoreStateAsync([update]);
+                    await _pullRequestUpdateState.SetReminderAsync(dueTimeInMinutes: 3);
+
+                    return ActionResult.Create(true, $"Pending updates applied. Branch {codeFlowStatus.PrBranch} not created yet");
+                }
+
+                // Step 3. Create a PR
+                string prUrl = await CreateCodeFlowPullRequestAsync(
+                    update,
+                    targetRepository,
+                    codeFlowStatus.PrBranch,
+                    targetBranch);
+
+                return ActionResult.Create(true, $"Pending updates applied. PR {prUrl} created");
+            }
+
+            // Technically, this should never happen as we create the code flow data before we even create the PR
```

Comment:
Having almost certain "dead code" makes this function more complicated. Even if this situation occurs, we'd need to kill the PR and remove the branch, etc. I am voting for removing :1134-1142 we relies on `codeFlowStatus` information during PR creating, for instance `codeFlowStatus.PrBranch`. So, I think it's safe to remove this validation.

Reply:
We had a similar situation occur a month ago when we renamed a namespace and were unable to parse the stored state. This led to Maestro being in an endless loop of crashing when processing a reminder.
I think it's kinda useful to have this block of code because
- It's not as long
- Self heals the state in cases like above
- Clearly signals that we're dependent on that data in this part of the processing pipeline